File: 15.py
#链表中倒数第k个节点
class ListNode(object):
	"""docstring for ListNode"""
	def __init__(self, value):
		#super(ListNode, self).__init__()
		self.next = None
		self.value = value
def kth_node(pnode, k):
	if not pnode or k==0:
		return False 
	phead1 = pnode
	phead2 = pnode
	count = k
	while count > 0 :
		if phead2.next:
			phead2 = phead2.next
			count -=1
		else:
			return False
	while phead2.next:
		phead1 = phead1.next
		phead2 = phead2.next
	return phead1
node1 = ListNode(10)
node2 = ListNode(11)
node3 = ListNode(13)
node1.next = node2
node2.next = node3
print(kth_node(node1,1).value)

File: 15.cpp
#include<iostream>
using namespace std;
struct ListNode{
	int value;
	ListNode* next;
};
ListNode* findKthToTial(ListNode* head,int k)
{   //headÎªNULL»òk=0
	if(head==NULL||k==0){
		return NULL;	
	}
	int length = 0;
    //k>length
	ListNode* pNode=head;
	ListNode* pBack=NULL;
	int count = 0;
	while(count<k-1){
		pNode = pNode->next;
		count++;
		
	}
	
	pBack=head;
	while(pNode->next!=NULL){
		pNode = pNode->next;
        pBack=pBack->next;
	}
	return pBack;
	
}
void main()
{
	ListNode* a = new ListNode();
	ListNode* b = new ListNode();
	ListNode* c = new ListNode();
	ListNode* d = new ListNode();
	ListNode* e = new ListNode();
	a->next = b;
	b->next = c;
	c->next = d;
	d->next = e;
	d->next = NULL;
	a->value =1;
	b->value =2;
	c->value =3;
	d->value =4;

    ListNode* k ;
	k= findKthToTial(a,3);
	cout<<k->value;
    unsigned int stop=0;
	cin>>stop;

}